stage_tester/st.py

```diff
@@ -148,6 +170,76 @@ def retrieve_cluster_list(organization, address, proxies, auth, verbose):
         print(cluster)
 
 
+def retrieve_results(address, proxies, auth, input_file, verbose):
+    """Retrieve results from the external data pipeline REST API endpoint."""
+    errors={}
+
+    # input file containing list of clusters
+    with open(input_file, "r") as input_file:
+        # iterate over all cluster names
+        for line in input_file:
+            cluster = line.strip()
+
+            if verbose:
+                print("Cluster: ", cluster)
+
+            # construct URL to get report for one specified cluster
+            url = '{}/v1/clusters/{}/report'.format(address, cluster)
+
+            if verbose:
+                print("URL to access:", url)
+
+            try:
+                # try to retrieve results for given cluster
+                retrieve_results_for_cluster(url, proxies, auth, cluster, verbose)
+            except Exception as e:
+                # store error to be used later
+                errors[cluster] = e
+
+    display_errors(errors)
+
+
+def display_errors(errors):
+    """Display all errors or exceptions thrown during the selected operation."""
+    print("-"*60)
+
+    if len(errors) > 0:
+        print("Errors detected during results processing")
+        for cluster, e in errors.items():
+            print(cluster, repr(e))
+    else:
+        print("No errors found")
+
+    print("-"*60)
+
+
+def retrieve_results_for_cluster(url, proxies, auth, cluster, verbose):
+    """Retrieve results for one specified cluster and store it into the file."""
+    # send request to REST API
+    response = requests.get(url, proxies=proxies, auth=auth)
+
+    # elementary check for response content
+    assert response is not None, "Proper response expected"
+    assert response.status_code == 200, "Unexpected HTTP code returned: {}".format(
```

Comment:
Doesn't have `requests` library some "constants" to use instead of "magic numbers"?

Reply:
hmm tbh I need to check, nice comment

Reply:
updated